interface/src/app/tribe_client/tribe_client.js

```diff
@@ -0,0 +1,39 @@
+angular.module('adage.tribe_client', [
+    'adage.tribe_client.resource',
+    'adage.tribe_client.directives',
+])
+
+.factory( 'UserFactory', ['User', function( User ) {
+    var promise = false;
+    var user = null;
+
+    return {
+        getUser: function() { //only call this when the promise completes
+            return user;
+        },
+        getPromise: function() {
+            if (!promise) {
+                promise = User.get({}, function( data ) {
+                    if (data.meta.total_count !== 0) {
+                        user = data.objects[0];
+                    }
+                });
+            }
+            return promise;
+        },
+        resetPromise: function() { // reset the promise in case we need to check the user again
+            promise = false;
```

Comment:
A `null` is interpreted as a "falsy" value, yes. That's why it made more sense to me to either have the promise object itself or `null`. You could define the `var promise = null;` in line 7 and reset it in line 25 and everything else should work the same.